src/controls/qml/LabeledActionButton.qml

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2023 - Timo Könnecke <github.com/eLtMosen>
+ * Copyright (C) 2022 - Ed Beroset <github.com/beroset>
+ * Copyright (C) 2020 - Darrel Griët <idanlcontact@gmail.com>
+ * Copyright (C) 2015 - Florent Revest <revestflo@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+
+Item {
+    // alias to recieve string icon.name
+    property alias icon: icon.name
+    // alias to recieve string label.text
+    property alias text: label.text
+    // left and right margin for the row content
+    property int rowMargin: Dims.w(15)
+    // size of the icon/s
+    property int iconSize: Dims.l(20)
+    // size of the label text
+    property int labelFontSize: Dims.l(6)
+    // onClicked dummy function can be overridden by caller
+    property var onClicked: function(){}
```

Comment:
Any reason not to use a `signal clicked` as is also used in the `ListItem`?

Reply:
Afaiu, `signal clicked` forwards the clicked signal to the parent that called the component.

But what we do is forward the onClicked action given by the parent to the component. 
Where it is again forwarded to the Highlight bar.
So that the highlight bar is the clickable item, but issues the onClicked command given in the parent.

Now writing it out, do you mean that it would work when the Highlightbar forwards the signal click to the LabeledActionButton, which then forwards the signal click to the parent, Where the onClicked command defined in the parent is then executed?

Reply:
I wasn't sure so I've checked this and it works fine.
These were the changes I made: https://github.com/MagneFire/qml-asteroid/commit/c49cc755abd237029cf83ae1054286632c7c10de

Reply:
Works, implemented the signal clicked way.